src/chainstate/stacks/boot/pox-2.clar

```diff
@@ -966,21 +997,34 @@
                     (pox-addr { version: (buff 1), hashbytes: (buff 20) })
                     (extend-count uint))
     (let ((stacker-info (stx-account stacker))
+          (stacker-state (get-stacker-info stacker))
           (amount-ustx (get locked stacker-info))
-          (unlock-height (get unlock-height stacker-info)))
+          (unlock-height (get unlock-height stacker-info))
+          (unlock-in-cycle (burn-height-to-reward-cycle unlock-height))
+          ;; if the account unlocks *during* this cycle (should only occur during testing),
+          ;; set first-extend-cycle to the next cycle.
+          (cur-cycle (current-pox-reward-cycle))
+          (first-extend-cycle (if (> (+ cur-cycle u1) unlock-in-cycle)
+                                     (+ cur-cycle u1) unlock-in-cycle))
+          ;; update stacker record
+          ;; maintaining valid stacking-state entries requires checking
+          ;;  whether there is an existing entry for the stacker in the state
+          ;; this would be the case if the stacker is extending a lockup from PoX-1
+          ;;  to PoX-2
+          (first-reward-cycle (match (get first-reward-cycle stacker-state)
+                                       ;; if stacker stacked in PoX2, then max(cur-cycle, stacker-state.first-reward-cycle) is valid
+                                       old-first-cycle (if (> cur-cycle old-first-cycle) cur-cycle old-first-cycle)
+                                       ;; otherwise, there aren't PoX2 entries until first-extend-cycle
+                                       first-extend-cycle)))
 
      ;; must be called with positive extend-count
      (asserts! (>= extend-count u1)
                (err ERR_STACKING_INVALID_LOCK_PERIOD))
 
-     (let ((unlock-in-cycle (burn-height-to-reward-cycle unlock-height))
-          ;; if the account unlocks *during* this cycle (should only occur during testing),
-          ;; set first-extend-cycle to the next cycle.
-           (cur-cycle (current-pox-reward-cycle))
-           (first-extend-cycle (if (> (+ cur-cycle u1) unlock-in-cycle)
-                                      (+ cur-cycle u1) unlock-in-cycle))
-           (last-extend-cycle  (- (+ first-extend-cycle extend-count) u1))
-           (lock-period (- last-extend-cycle cur-cycle))
+     (let ((last-extend-cycle  (- (+ first-extend-cycle extend-count) u1))
+           ;; this seems incorrect
```

Comment:
:eyes:

Reply:
That comment applies to the commented out calculation.

Reply:
Remove this and the commented out calculation.